sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+
+        return base64.b64encode(pickle.dumps(self._initial_response)).decode("ascii")
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token: str, **kwargs) -> Tuple[Any, Any, Callable]:
+        try:
+            client = kwargs["client"]
+        except KeyError:
+            raise ValueError("Need kwarg 'client' to be recreated from continuation_token")
+
+        import pickle
+
+        initial_response = pickle.loads(base64.b64decode(continuation_token))  # nosec
+        # Restore the transport in the context
+        initial_response.context.transport = client._client._pipeline._transport  # pylint: disable=protected-access
+        return client, initial_response, lambda: None
+```
+
+And now, to plug into the client code:
+
+```python
+class ServiceOperations:
+
+    def begin_upload(self, data: AnyStr, **kwargs) -> LROPoller[JSON]:
+        continuation_token = kwargs.pop("continuation_token", None)
+        polling_method = CustomLROBasePolling(**kwargs)
+        if continuation_token is not None:
+            return LROPoller.from_continuation_token(
+                continuation_token=continuation_token,
+                polling_method=polling_method,
+                client=self
+            )
+        response = self._client.create_upload(data, **kwargs)
+
+        return LROPoller(
+            client=self,
+            initial_response=response,
+            deserialization_callback=lambda: None,
+            polling_method=polling_method,
+        )
+```
+
+Note that we need to account for a `continuation_token` being passed by the user, in which case we should not make the
+initial call again, but rather resume polling from the rehydrated state.
+
+### Poller - LROPoller/AsyncLROPoller
+
+The last option is if you need to customize the public interface of the `LROPoller` / `AsyncLROPoller`.
+
+Example: I want to add a cancel method to my poller
+
+Note that if you support rehydration of the LRO via `continuation_token`, you must override the `from_continuation_token` 
+method so that the custom poller is used.
+
+```python
+class CustomLROPoller(LROPoller[PollingReturnType]):
+
+    def cancel(self, **kwargs) -> None:
+        """Cancel the upload"""
+        return self.polling_method().client.cancel_upload_file(self.polling_method().file_id, **kwargs)
+
+    @classmethod
+    def from_continuation_token(
+        cls, polling_method: PollingMethod[PollingReturnType], continuation_token: str, **kwargs
+    ) -> "CustomLROPoller[PollingReturnType]":
+        (
+            client,
+            initial_response,
+            deserialization_callback,
+        ) = polling_method.from_continuation_token(continuation_token, **kwargs)
+        return cls(client, initial_response, deserialization_callback, polling_method)
+
+```
+
+And now, to plug into the client code:
+
+```python
+class ServiceOperations:
+
+    def begin_upload(self, data: AnyStr, **kwargs) -> CustomLROPoller[JSON]:
```

Comment:
Maybe add a note about the updated return type here - we should only need an explicitly type reference _if_ new public API has been added. And if that's the case, that new Poller object will need to be added to the package `__init__.py` so that the new public API will be properly documented.